### PR TITLE
Feat/axios api

### PR DESCRIPTION
**Pull request by castrocami**
### Contexto
Reconstruyendo Lakatan. [En este pitch](https://www.notion.so/platanus/Interfaz-de-equipos-v1-0-1c4cd55247ce4ec5b0d94dad66cc294a), que corresponde a CRUDs de equipos
​
### Qué se esta haciendo
- Crear metodos para hacer display de teams consumiendo teams api
- Se eliminó el display de teams mediante json
​
#### En particular hay que revisar
​

-----------

### CODE REVIEW

File: app/javascript/api/index.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        error.details = [`Error interno del servidor: ${data.detail}`];
+        break;
+      default:
+        break;
+      }
+    }
+
+    return Promise.reject(error);
+  },
+);
```
Comment (aaguilarg) [inline app/javascript/api/index.js:37]: Borremos esto mejor

File: app/javascript/api/teams.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import api from './index';
+
+function getTeams({ organizationId }) {
+  return api({
+    method: 'get',
+    url: `/api/v1/organizations/${organizationId}/teams`,
+  });
+}
+
+export default getTeams;
```
Comment (aaguilarg) [inline app/javascript/api/teams.js:10]: Se ve bien, eso si un detalle. Como vamos a seguir metiendo metodos acá, dejemos algo asi
```javascript
export default {
  getTeams({ organizationId }) {
    return api({
      method: 'get',
      url: `/api/v1/organizations/${organizationId}/teams`,
  });
};
```

y lo importamos asi 
```javascript
import TeamsApi from 'ruta'
```
y lo usamos asi

```javascript
TeamsApi.getTeams(...)...
```

**Review by aaguilarg — APPROVED** [review]

**Review by aaguilarg — CHANGES_REQUESTED** [review]
deje unos comentarios para arreglar

File: app/javascript/components/teams/teams.vue [hunk 3]
```diff
@@ -8,16 +8,29 @@
       class="p-6 mb-4 border rounded-lg"
       :key="team.id"
     >
-      {{ team.name }}
+      {{team.attributes.name}}
```
Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:11]: ⚠️ **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expected 1 space after '{{', but not found   vue/mustache-interpolation-spacing

Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:11]: ⚠️ **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Expected 1 space before '}}', but not found  vue/mustache-interpolation-spacing

File: app/javascript/components/teams/teams.vue [hunk 4]
```diff
@@ -8,16 +8,29 @@
...
     </div>
   </div>
 </template>
 
 <script>
+import TeamsApi from '../../api/teams';
```
Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:17]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Trailing spaces not allowed                  no-trailing-spaces

File: app/javascript/components/teams/teams.vue [hunk 5]
```diff
@@ -8,16 +8,29 @@
...
 
 <script>
+import TeamsApi from '../../api/teams'; 
+
 export default {
   props: {
     organization: { type: Object, required: true },
-    teams: { type: Array, required: true },
   },
+  data(){
```
Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:23]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Missing space before opening brace           space-before-blocks

File: app/javascript/components/teams/teams.vue [hunk 6]
```diff
@@ -8,16 +8,29 @@
...
+    return{
```
Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:24]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Expected space(s) after "return"             keyword-spacing

File: app/javascript/components/teams/teams.vue [hunk 7]
```diff
@@ -8,16 +8,29 @@
...
+      teams: [],
+    }
```
Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:26]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Missing semicolon                            semi

File: app/javascript/components/teams/teams.vue [hunk 8]
```diff
@@ -8,16 +8,29 @@
...
+  },
+  mounted() {
+    TeamsApi.getTeams({ organizationId: this.organization.id }).then((response)=>{
```
Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:29]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Missing space before =>                      arrow-spacing

Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:29]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Missing space after =>                       arrow-spacing

File: app/javascript/components/teams/teams.vue [hunk 9]
```diff
@@ -8,16 +8,29 @@
...
+      this.teams = response.data.data;
+    }).catch((error)=>{
```
Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:31]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Missing space before =>                      arrow-spacing

Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:31]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Missing space after =>                       arrow-spacing

File: app/javascript/components/teams/teams.vue [hunk 10]
```diff
@@ -8,16 +8,29 @@
...
+      console.error('Error obtaining teams from api', error)
```
Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:32]: ⚠️ **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected console statement                 no-console

Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:32]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Missing semicolon                            semi

File: app/javascript/components/teams/teams.vue [hunk 11]
```diff
@@ -8,16 +8,29 @@
...
+    });
```
Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:33]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Trailing spaces not allowed                  no-trailing-spaces

File: app/javascript/components/teams/teams.vue [hunk 12]
```diff
@@ -8,16 +8,29 @@
...
+      teams: [],
+    }
+  },
+  mounted() {
+    TeamsApi.getTeams({ organizationId: this.organization.id }).then((response)=>{
+      this.teams = response.data.data;
+    }).catch((error)=>{
+      console.error('Error obtaining teams from api', error)
+    }); 
+  }
```
Comment (monkeyci) [inline app/javascript/components/teams/teams.vue:34]: 🚫 **[eslint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>  Missing trailing comma                       comma-dangle

**Review by aaguilarg — APPROVED** [review]

**Pull request merged**